### PR TITLE
Refactor EventLoop singleton to namespace methods

### PR DESCRIPTION
**Pull request by fuegas**
A few things changed in this commit:
- Component renamed from genericeventhandler to event_handlers
The component had generic in its name, but none of the classes beared it, so I removed the "generic" part of the name. Secondly, it was no event loop, it was a list of event handlers, so a simpler name for the class was EventHandlers.

- Refactored singleton class to namespace
The code using this components should not be bothered if the implementation uses a singleton, a factory, or whatever. Using get_instance limits you to refactor what happens internally as you're somewhat exposing your internals. When the instance methods (add, remove, handle for simplicity) are made available at static level you're almost at the same point as using a namespace with methods. Because the list `handlers` is declared in the namespace and in the cpp file, others can't access it or add new handlers suddenly.

- Renamed event_handler to handle_event
The action requested is to handle an event, so I've renamed the method to reflect it being an action.

### CODE REVIEW

File: main/main.cpp [hunk 1]
```diff
@@ -4,20 +4,20 @@
 
 #include "sdkconfig.h"
 #include "esp_system.h"
-#include "esp_event_loop.h"
```
Comment (jmvdwerf) [inline main/main.cpp:7]: This include is required for the call  to ```esp_event_loop_init``` in line 33

**Review by jmvdwerf — CHANGES_REQUESTED** [review]
There is a problem in building, as you did not change the CMakeLists.txt of the wificontroller component  (```components/wificontroller/CMakeLists.txt```) Here ```genericeventhandler``` should be changed in ```event_handlers``` as this is a reference to that component...

Note that in main.cpp the ```#include "esp_event_loop.h"``` is used for initializing the loop, hence, the include is required...

With these two changes, the build works again...

File: main/main.cpp (hunk 1 above)
Comment (fuegas) [reply main/main.cpp:7]: The contents of `esp_event_loop.h` is:
```c++
#pragma once
#warning "esp_event_loop.h is deprecated, please include esp_event.h instead"
#include "esp_event.h"
```
So I thought it was save to alter it 😉

Comment (jmvdwerf) [reply main/main.cpp:7]: Solved, as we move to the release/v4.0 version 😄

**Review by jmvdwerf — APPROVED** [review]
It builds and runs also on my device!

**Pull request merged**